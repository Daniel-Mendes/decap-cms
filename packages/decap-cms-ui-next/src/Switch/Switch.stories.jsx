import React from 'react';
import { action } from '@storybook/addon-actions';
import { useArgs } from '@storybook/preview-api';

import Switch from '.';

export default {
  title: 'Components/Switch',
  component: Switch,
  argTypes: {
    checked: {
      control: 'boolean',
      table: {
        defaultValue: { summary: false },
      },
    },
  },
  args: {
    checked: false,
  },
};

export function _Switch(args) {
  const [{ checked }, updateArgs] = useArgs();

  function toggleChecked() {
    updateArgs({ checked: !checked });
  }

  return (
    <Switch
      {...args}
      onCheckedChange={event => {
        action('onChange')(event ? 'checked' : 'unchecked');
        toggleChecked();
      }}
    />
  );
}
