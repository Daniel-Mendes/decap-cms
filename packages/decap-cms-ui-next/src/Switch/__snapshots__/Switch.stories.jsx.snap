// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Switch _Switch smoke-test 1`] = `"<input type="checkbox" class="css-1rg9wi1-StyledInput e1lye9la1"><button class="css-1407tdr-SwitchButton e1lye9la0"></button>"`;
